submit-api/src/Cardano/TxSubmit.hs

```diff
@@ -32,16 +33,20 @@ runTxSubmitWebapi :: TxSubmitNodeParams -> IO ()
 runTxSubmitWebapi tsnp = do
     tsnc <- readTxSubmitNodeConfig (unConfigFile $ tspConfigFile tsnp)
     trce <- mkTracer tsnc
-    tsv <- newTxSubmitVar
-    Async.race_
-      (runTxSubmitNode tsv trce tspProtocol tspNetworkId tspSocketPath)
-      (runTxSubmitServer tsv trce (tspWebPort tsnp))
-    logInfo trce "runTxSubmitWebapi: Async.race_ returned"
+    (metrics, metricsServer) <- registerMetricsServer
+    txSubmitServer <- Async.async $
+      runTxSubmitServer trce metrics tspWebPort tspProtocol tspNetworkId tspSocketPath
+    void $ Async.waitAnyCancel
+      [ txSubmitServer
+      , metricsServer
+      ]
```

Comment:
Just out of curiosity, why not just use `Async.race_` here? (Since the result is ignored.)

Reply:
I just went with `Async.waitAnyCancel` because `metricServer` is an `Async ()`. 

I could do something like `Async.race_ (runTxSubmitServer trce metrics tspWebPort tspProtocol tspNetworkId tspSocketPath) (wait metricsServer)` instead of what I did with `Async.waitAnyCancel`, if that's preferred.

Reply:
> I just went with `Async.waitAnyCancel` because `metricsServer` is an `Async ()`.

Fair enough! On reflection, I agree that using `waitAnyCancel` makes perfect sense here.